### PR TITLE
Fix CMake.test() target for Ninja Multi-Config

### LINKED ISSUE #11405: [bug] `cmake.test()` cannot execute
### Environment Details (include every applicable attribute)
  * Operating System+version: Ubuntu 22.04 on WSL
  * Compiler+version: CMake 3.23.1
  * Conan version: 1.49.0
  * Python version: 3.10.4

### Steps to reproduce (Include if Applicable)

Put `cmake.test()` and run `conan build .` with `Ninja Multi-Config` as a generator.

### Logs (Executed commands with output) (Include/Attach if Applicable)

```
conanfile.py (mp-units/0.8.0): CMake command: cmake --build "/home/mpusz/repos/units/build-gcc-11" --config Release '--' '-j12'
[308/308] Linking CXX executable test/unit_test/runtime/Release/unit_tests_runtime
conanfile.py (mp-units/0.8.0): CMake command: cmake --build "/home/mpusz/repos/units/build-gcc-11" --config Release '--target' 'RUN_TESTS' '--' '-j12'
ninja: error: unknown target 'RUN_TESTS'
ERROR: conanfile.py (mp-units/0.8.0): Error in build() method, line 172
        cmake.test()
        ConanException: Error 1 while executing cmake --build "/home/mpusz/repos/units/build-gcc-11" --config Release '--target' 'RUN_TESTS' '--' '-j12'
```

CMake target `RUN_TESTS` is not available. `test` target works correctly but Conan does not use it for some reason.

### PR DESCRIPTION
**Pull request by Naufragous**
Changelog: Bugfix: Use correct build target in `CMake.test()` for the Ninja Multi-Config generator.
Docs: omit
Fixes #11405

### CODE REVIEW

**Comment by Naufragous** [comment]
I used the [test_run_tests](https://github.com/conan-io/conan/blob/4ed1bee0fb81b2826208e8c1c824c99fb6d69be8/conans/test/unittests/client/build/cmake_test.py#L1109) test case from _conans/test/unittests/client/build/cmake_test.py_ as basis for this new test. Unfortunately there seem to be not that many tests for the new CMake tool. Please advise if something is off.

File: conans/test/unittests/tools/cmake/test_cmake_test.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    conanfile = ConanFileMock()
+    conanfile.conf = Conf()
+    conanfile.folders.generators = "."
+    conanfile.folders.set_base_generators(temp_folder())
+    conanfile.settings = settings
+
+    write_cmake_presets(conanfile, "toolchain", generator, {})
+    cmake = CMake(conanfile)
+    cmake.test()
+    assert "'--target' '%s'" % target in conanfile.command
```
Comment (memsharded) [inline conans/test/unittests/tools/cmake/test_cmake_test.py]:
```suggestion
    if platform.system() == "Windows":
        assert "--target %s" % target in conanfile.command
    else:
        assert "'--target' '%s'" % target in conanfile.command
```

There is a small difference atm in the command in Windows, the args are not quoted.

Comment (memsharded) [reply conans/test/unittests/tools/cmake/test_cmake_test.py]: cc/ @lasote This is one of the things related to https://github.com/conan-io/conan/pull/11553, that seems a bit weird to me, that the commands quoting is different in different systems.

**Review by memsharded — COMMENTED** [review]
Thanks for contributing this fix.

File: conans/test/unittests/tools/cmake/test_cmake_test.py (hunk 1 above)
Comment (Naufragous) [reply conans/test/unittests/tools/cmake/test_cmake_test.py]: Thank you for the suggestion. I added the `platform` import and made the search pattern conditional on the platform.

**Review by memsharded — APPROVED** [review]

**Comment by memsharded** [comment]
Merged, this will be released in 1.51.
Thanks again for your contribution!

**Pull request merged**